app/controllers/admin/product_import_controller.rb

```diff
@@ -126,6 +127,16 @@ def model_class
       ProductImport::ProductImporter
     end
 
+    def mktmpdir
+      tmpdir = tmpdir_base + SecureRandom.hex(6)
+      Dir.mkdir(tmpdir)
+      tmpdir
+    end
+
+    def tmpdir_base
+      Rails.root.join('tmp', TMPDIR_PREFIX).to_s
+    end
```

Comment:
The knowledge of the path construction is in multiple places now as well. Instead of having a directory name prefix, we could just create a directory `tmp/product_import/` and then have random filenames within. Would that make more sense?

Reply:
Yeah that would probably make more sense, but we would need to re-implement without `Dir::Tmpname.create` (which handles name collisions for us).
Hmm, I guess that's what `make_tmpname` mentioned in https://github.com/openfoodfoundation/openfoodnetwork/pull/3435 is for, but that method was abandoned on that PR. I forget why, but as you say I don't think we should put more energy in ;)